scheduler/scheduler.go

```diff
@@ -337,17 +342,73 @@ func (s *scheduler) createTask(sch schedule.Schedule, wfMap *wmap.WorkflowMap, s
 		return nil, te
 	}
 
+	// subscribedPluginAsserts includes rules that need to be evaluated once we
+	// have mapped the metrics to specific collector plugins.  Examples include
+	// asserting that streaming tasks don't reference non-streaming collectors.
+	subscribedPluginAsserts := []core.SubscribedPluginAssert{}
 	// Group dependencies by the node they live on
 	// and validate them.
 	depGroups := getWorkflowPlugins(wf.processNodes, wf.publishNodes, wf.metrics)
 	for k, group := range depGroups {
+
+		// populate subscribedPluginAsserts
+		switch sch.(type) {
+		case *schedule.StreamingSchedule:
+			// assert no non-streaming plugins
+			subscribedPluginAsserts = append(subscribedPluginAsserts, func(plugins []core.SubscribedPlugin) serror.SnapError {
+				for _, plg := range plugins {
+					if plg.TypeName() != plugin.StreamCollectorPluginType.String() {
+						return serror.New(
+							ErrPluginIncompatibleWithScheduleType,
+							map[string]interface{}{
+								"schedule_type": fmt.Sprintf("%T", sch),
+								"plugin_name":   plg.Name(),
+								"plugin_type":   plg.TypeName(),
+							},
+						)
+					}
+				}
+				return nil
+			})
+			// assert only a single streaming plugin
+			subscribedPluginAsserts = append(subscribedPluginAsserts, func(plugins []core.SubscribedPlugin) serror.SnapError {
+				if len(plugins) > 1 {
+					return serror.New(
+						ErrMultipleStreamingPlugins,
+						map[string]interface{}{
+							"schedule_type":     fmt.Sprintf("%T", sch),
+							"num_of_collectors": len(plugins),
+						},
+					)
+				}
+				return nil
+			})
+		default:
+			// assert no streaming plugins
+			subscribedPluginAsserts = append(subscribedPluginAsserts, func(plugins []core.SubscribedPlugin) serror.SnapError {
+				for _, plg := range plugins {
+					if plg.TypeName() == plugin.StreamCollectorPluginType.String() {
+						return serror.New(
+							ErrPluginIncompatibleWithScheduleType,
+							map[string]interface{}{
+								"schedule_type": fmt.Sprintf("%T", sch),
+								"plugin_name":   plg.Name(),
+								"plugin_type":   plg.TypeName(),
```

Comment:
I did not receive those "schedule_type", "plugin_name", etc. in error message. 
What I received for test-scenario of using "simple" schedule for metrics coming from stream collector plugin is: 
```
$ snaptel task create -t simple.json
Using task manifest to create task
Error creating task:Plugin is incompatible with the tasks schedule type.
```
```
$ curl -vX POST localhost:8181/v1/tasks -d @simple.json --header "Content-Type: application/json"
{
  "meta": {
    "code": 500,
    "message": "Plugin is incompatible with the tasks schedule type.",
    "type": "error",
    "version": 1
  },
  "body": {
    "message": "Plugin is incompatible with the tasks schedule type.",
    "fields": {}
  }
* Connection #0 to host localhost left intact
}
```